### PR TITLE
 Add docs to flytekit-java module #patch

### PR DESCRIPTION
**Pull request by narape**
# TL;DR
Add documentation for public classes in the flytekit-java

## Type
 - [x] Documentation
 - [ ] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [ ] Code completed
 - [ ] Smoke tested
 - [ ] Unit tests added
 - [x] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
For public classes or interfaces:
* comment class and every public method or constant
* protected methods that are meant to be extended by users.
For package private classes or interfaces.
* Just comment add a class comment.

There are also some code modernization by using newer Java 11 API.
And finally SdkNode.getOuput("foo") was removed

## Tracking Issue
_NA_
## Follow-up issue
_NA_

### CODE REVIEW

**Comment by pablocasares** [comment]
LGTM

File: flytekit-java/src/main/java/org/flyte/flytekit/MoreCollectors.java [hunk 1]
```diff
@@ -1,43 +0,0 @@
...
-import static java.util.stream.Collectors.toList;
-import static java.util.stream.Collectors.toMap;
-
-import java.util.Collections;
-import java.util.List;
-import java.util.Map;
-import java.util.function.Function;
-import java.util.stream.Collector;
-
-class MoreCollectors {
```
Comment (narape) [inline flytekit-java/src/main/java/org/flyte/flytekit/MoreCollectors.java:29]: **Note to reviewers**
Java 11 Collectors has these methods, so we use the standard ones

File: flytekit-java/src/main/java/org/flyte/flytekit/SdkNode.java [hunk 2]
```diff
@@ -30,33 +30,43 @@ protected SdkNode(SdkWorkflowBuilder builder) {
...
   public abstract Map<String, SdkBindingData<?>> getOutputBindings();
 
+  /**
+   * Returns output of this node as an {@link OutputT}.
+   *
+   * @return output.
+   */
   public abstract OutputT getOutputs();
 
-  public SdkBindingData<?> getOutput(String name) {
```
Comment (narape) [inline flytekit-java/src/main/java/org/flyte/flytekit/SdkNode.java:37]: ** Note to reviewers ** 
This method became obsolete with typed outputs

File: flytekit-java/src/main/java/org/flyte/flytekit/SdkStruct.java [hunk 3]
```diff
@@ -27,69 +27,146 @@
...
 
+    /**
+     * Adds a list of flyte float field.
+     *
+     * @param name name of the field
+     * @param values values to set
+     * @return this builder
+     */
     public Builder addFloatCollectionField(String name, List<Double> values) {
-      return this.<Double>addListValueField(name, values, Value::ofNumberValue);
```
Comment (narape) [inline flytekit-java/src/main/java/org/flyte/flytekit/SdkStruct.java:75]: **Note to reviewers**
Java 11 has better type inference than Java 8 so it is not longer necessary to specify the type here

File: flytekit-java/src/test/java/org/flyte/flytekit/SdkWorkflowBuilderTest.java [hunk 4]
```diff
@@ -246,26 +246,6 @@ void testDuplicateNodeId() {
         e.getMessage());
   }
 
-  @ParameterizedTest
-  @MethodSource("createTransform")
-  void testVariableNameNotFound_output(
```
Comment (narape) [inline flytekit-java/src/test/java/org/flyte/flytekit/SdkWorkflowBuilderTest.java:251]: **Note to reviewers**
This test became invalid now the `node1.getOutput("foo")`  was removed

**Review by andresgomezfrr — APPROVED** [review]

**Review by pablocasares — APPROVED** [review]
